tests/worldmap/QuadTreeTest.cpp

```diff
@@ -0,0 +1,233 @@
+#include "../../src/worldmap/QuadTree.h"
+
+#include <iostream>
+
+#include <catch2/catch.hpp>
+
+namespace
+{
+double randNum(double low, double high)
+{
+	return low + (rand() / (RAND_MAX / (high - low))); // NOLINT(cert-msc50-cpp)
+}
+
+point_t randPoint(double size)
+{
+	return {randNum(-size / 2, size / 2), randNum(-size / 2, size / 2), 1};
```

Comment:
It might be worth thinking about what will happen when we move to 3D. We can always probably project obstacle points down into a 2D map, but maybe at some point we will want to build a 3D map. No changes needed in this PR, but just food for thought.

Reply:
Hmm, in that case I'd either have to overhaul all the lidar map code to work in 3d, or there's probably a library out there well suited for this. Could we use this for now, and I'll do some research into 3d?

Reply:
Yeah, of course this is fine for now. 3D research might be a bit premature at this point, but feel free to take a look if that sounds interesting